reputation/src/ars.rs

```diff
@@ -0,0 +1,184 @@
+//! Active Reputation Set
+
+use queues::{CircularBuffer, IsQueue};
+
+use crate::trs::{decrement_cache, increment_cache};
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+};
+
+/// Active Reputation Set
+///
+/// This data structure keeps track of every identity `K` in a buffer of
+/// time. In order to keep track the identities contained in the buffer,
+/// these are stored in a circular queue (FIFO).
+///
+/// The method `push_activity` insert a vector of identities in the structure
+/// and also update with the identities that are expired.
+
+#[derive(Debug)]
+pub struct ActiveReputationSet<K>
+where
+    K: Clone + Eq + Hash,
+{
+    // A cache of <identity: activity>
+    // All the identities with activity are in the cache
+    map: HashMap<K, u16>,
+    // The list of active identities ordered by time
+    queue: CircularBuffer<HashSet<K>>,
+}
+
+impl<K> ActiveReputationSet<K>
+where
+    K: Clone + Eq + Hash,
+{
+    /// Default `ActiveReputationSet<K>` initializer
+    ///
+    /// # Returns
+    /// A new, empty `ActiveReputationSet<K>`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use witnet_reputation::ActiveReputationSet;
+    /// let ars: ActiveReputationSet<isize> = ActiveReputationSet::new(3);
+    /// assert_eq!(ars.buffer_size(), 0);
+    /// assert_eq!(ars.capacity(), 3);
+    /// ```
+    pub fn new(capacity: usize) -> Self {
+        Self {
+            map: HashMap::new(),
+            queue: CircularBuffer::new(capacity),
+        }
+    }
+
+    /// Gets the capacity of the `ActiveReputationSet<K>`
+    pub fn capacity(&self) -> usize {
```

Comment:
For consistency with `buffer_size()`

```suggestion
    pub fn buffer_capacity(&self) -> usize {
```